node/pkg/p2p/p2p.go

```diff
@@ -247,9 +247,35 @@ func Run(
 			panic(err)
 		}
 
+		var sub *pubsub.Subscription
+		var th *pubsub.Topic
+
+		go func() {
```

Comment:
How will setting this up here interact if something fails below and the function returns? Would it be better (and less lines of code) to put this cleanup at the end of the function?

Reply:
I changed it to have defers instead of waiting on ctx.Done in a go-routine. I also split it up into multiple defers right after the resource creation, which I think is the more common practice. Lmk what you think.

Reply:
don't immediately see why the defers wouldn't work and I don't recall if there was a reason I hadn't done that in another script.